core/src/main/java/org/apache/iceberg/mapping/MappedFields.java

```diff
@@ -37,21 +38,21 @@ public static MappedFields of(List<MappedField> fields) {
   }
 
   private final List<MappedField> fields;
-  private final Map<String, Integer> nameToId;
-  private final Map<Integer, MappedField> idToField;
+  private transient Map<String, Integer> nameToId;
+  private transient Map<Integer, MappedField> idToField;
 
   private MappedFields(List<MappedField> fields) {
     this.fields = ImmutableList.copyOf(fields);
-    this.nameToId = indexIds(fields);
-    this.idToField = indexFields(fields);
+    lazyNameToId();
```

Comment:
I kept the initialization in place. It triggers validation. Otherwise, some tests fail.